adcc/ElectronicTransition.py

```diff
@@ -0,0 +1,269 @@
+            "rotational_strength" (ECD spectrum with rotational strength)
+        width : float, optional
+            Gaussian broadening standard deviation or Lorentzian broadening
+            gamma parameter. The value should be given in atomic units
+            and will be converted to the unit of the energy axis.
+        """
+        if xaxis == "eV":
+            eV = constants.value("Hartree energy in eV")
+            energies = self.excitation_energy * eV
+            width = width * eV
+            xlabel = "Energy (eV)"
+        elif xaxis in ["au", "Hartree", "a.u."]:
+            energies = self.excitation_energy
+            xlabel = "Energy (au)"
+        elif xaxis == "nm":
+            hc = constants.h * constants.c
+            Eh = constants.value("Hartree energy")
+            energies = hc / (self.excitation_energy * Eh) * 1e9
+            xlabel = "Wavelength (nm)"
+            if broadening is not None and not callable(broadening):
+                raise ValueError("xaxis=nm and broadening enabled is "
+                                 "not supported.")
+        elif xaxis in ["cm-1", "cm^-1", "cm^{-1}"]:
+            towvn = constants.value("hartree-inverse meter relationship") / 100
+            energies = self.excitation_energy * towvn
+            width = width * towvn
+            xlabel = "Wavenumbers (cm^{-1})"
+        else:
+            raise ValueError("Unknown xaxis specifier: {}".format(xaxis))
+
+        if yaxis in ["osc", "osc_strength", "oscillator_strength", "f"]:
+            absorption = self.oscillator_strength
+            ylabel = "Oscillator strengths (au)"
+        elif yaxis in ["dipole", "dipole_norm", "μ"]:
+            absorption = np.linalg.norm(self.transition_dipole_moment, axis=1)
+            ylabel = "Modulus of transition dipole (au)"
+        elif yaxis in ["cross_section", "σ"]:
+            absorption = self.cross_section
+            ylabel = "Cross section (au)"
+        elif yaxis in ["rot", "rotational_strength", "rotatory_strength"]:
+            absorption = self.rotatory_strength
+            ylabel = "Rotatory strength (au)"
+        else:
+            raise ValueError("Unknown yaxis specifier: {}".format(yaxis))
+
+        sp = ExcitationSpectrum(energies, absorption)
+        sp.xlabel = xlabel
+        sp.ylabel = ylabel
+        if not broadening:
+            plots = sp.plot(style="discrete", **kwargs)
+        else:
+            kwdisc = kwargs.copy()
+            kwdisc.pop("label", "")
+            plots = sp.plot(style="discrete", **kwdisc)
+
+            kwargs.pop("color", "")
+            sp_broad = sp.broaden_lines(width, shape=broadening)
+            plots.extend(sp_broad.plot(color=plots[0].get_color(),
+                                       style="continuous", **kwargs))
+
+        if xaxis in ["nm"]:
+            # Invert x axis
+            plt.xlim(plt.xlim()[::-1])
+        return plots
+
+
+class State2StateTransition(ElectronicTransition):
```

Comment:
I'd shorten it to State2States (similar to ExcitedStates)

Reply:
Done.